lib/pbench/test/unit/server/query_apis/test_datasets_list.py

```diff
@@ -12,69 +12,37 @@ class TestDatasetsList:
-        """
-        Test behavior when Authorization header does not have access to other user's data
-        """
-        # The pbench_token fixture logs in as user "drb"
-        # Trying to access the data belong to the user "pp"
-        response = client.post(
-            f"{server_config.rest_uri}/datasets/list",
-            headers={"Authorization": "Bearer " + pbench_token},
-            json={
-                "user": "pp",
-                "controller": "cpntroller.name",
-                "start": "2020-08",
-                "end": "2020-10",
-            },
-        )
-        assert response.status_code == HTTPStatus.FORBIDDEN
-
-    @pytest.mark.parametrize(
-        "user", ("drb", "pp"),
-    )
-    def test_accessing_data_with_invalid_token(
-        self, client, server_config, pbench_token, user
-    ):
-        """
-        Test behavior when Authorization header does not have access to other user's data
-        """
-        # valid token logout
-        response = client.post(
-            f"{server_config.rest_uri}/logout",
-            headers=dict(Authorization="Bearer " + pbench_token),
-        )
-        assert response.status_code == 200
-        response = client.post(
-            f"{server_config.rest_uri}/datasets/list",
-            headers={"Authorization": "Bearer " + pbench_token},
-            json={"user": user, "start": "2020-08", "end": "2020-10"},
-        )
-        assert response.status_code == HTTPStatus.FORBIDDEN
-
-    def test_missing_json_object(self, client, server_config, pbench_token):
-        """
-        Test behavior when no JSON payload is given
-        """
-        response = client.post(
-            f"{server_config.rest_uri}/datasets/list",
-            headers={"Authorization": "Bearer " + pbench_token},
-        )
-        assert response.status_code == HTTPStatus.BAD_REQUEST
-        assert response.json.get("message") == "Invalid request payload"
+    @pytest.fixture(autouse=True)
+    def _setup(self):
+        self.endpoint = "/datasets/list"
+        self.json = {
+            "user": "drb",
+            "controller": "cpntroller.name",
+            "start": "2020-08",
+            "end": "2020-10",
+        }
+        self.json_bad_date = {
+            "user": "drb",
+            "controller": "dbutenho.csb",
+            "start": "2020-12",
+            "end": "2020-19",
+        }
+        self.json_error = {
+            "user": "",
+            "controller": "foobar",
+            "start": "2020-08",
+            "end": "2020-08",
+        }
+        # self.empty_response_payload = {
+        #     "took": 1,
+        #     "timed_out": False,
+        #     "_shards": {"total": 1, "successful": 1, "skipped": 0, "failed": 0},
+        #     "hits": {
+        #         "total": {"value": 0, "relation": "eq"},
+        #         "max_score": None,
+        #         "hits": [],
+        #     },
+        # }
```

Comment:
Did I miss some default? Why is this missing? (And don't leave it commented out, in any case.)

Reply:
Yes. It was not present in the original `test_datasets_list` so I just wondered if it was intentionally left out, but in any case i removed it.